sigpy/linop.py

```diff
@@ -46,9 +46,10 @@ class Linop():
         oshape: output shape.
         ishape: input shape.
         H: adjoint linear operator.
+        N: normal linear operator.
 
     """
-    def __init__(self, oshape, ishape, repr_str=None):
+    def __init__(self, oshape, ishape, repr_str=None, adj=None, normal=None):
```

Comment:
why do you want to add adj and normal as arguments? The adjoint operator used to be defined through _adjoint_linop. Can you extend it to normal with _normal_linop as well?

Reply:
I passed it as an argument to avoid repeatedly calling constructors. Taking a Fourier linop F as an example,  the user can implement `_adjoint_linop` to return F.H that has `adj=self` passed in as the constructor.

Reply:
However, if you prefer calling a new constructor each time, I can revert.

Reply:
Yea, can you keep the old interface for this request? just to keep things separate. thanks

Reply:
Sounds good. Do you want me to implement the above as a new PR or just bin it?

Reply:
yea splitting that as a new PR would be good. Coz it'll be good to have interface consistent with Prox too